### PR TITLE
Review HRN::IsNull function

### PR DESCRIPTION
**Pull request by mykhailo-kuchma**
Add unit tests for HRN class
Adapt Tokenizer to google coding style
Remove unused functionality from Tokenizer
Remove handling of http and https protocol.

Relates-To: OLPEDGE-1201

Signed-off-by: Mykhailo Kuchma <ext-mykhailo.kuchma@here.com>

### CODE REVIEW

File: olp-cpp-sdk-core/src/client/HRN.cpp [hunk 1]
```diff
@@ -51,62 +51,54 @@ std::string HRN::ToString() const {
...
-         this->catalogId;
+         catalogId;
 }
 
 HRN::HRN(const std::string& input) {
   Tokenizer tokenizer(input, ':');
 
   // hrn must start with "hrn:"
-  if (!tokenizer.hasNext()) return;
+  if (!tokenizer.HasNext()) return;
```
Comment (andescu) [inline olp-cpp-sdk-core/src/client/HRN.cpp]: I would add braces.
Same below.

File: olp-cpp-sdk-core/src/client/Tokenizer.h [hunk 2]
```diff
@@ -27,49 +27,39 @@ namespace client {
...
-      : str(input), pos(0), sep(separator) {}
+      : str_(input), pos_(0), sep_(separator) {}
 
-  bool hasNext() const { return !str.empty() && (pos != std::string::npos); }
+  bool HasNext() const { return !str_.empty() && (pos_ != std::string::npos); }
 
-  std::string next() {
-    if (pos == std::string::npos) return std::string();
+  std::string Next() {
+    if (pos_ == std::string::npos) return std::string();
```
Comment (andescu) [inline olp-cpp-sdk-core/src/client/Tokenizer.h]: I would add braces.
Same below.

File: olp-cpp-sdk-core/tests/client/HRNTest.cpp [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+      EXPECT_FALSE(hrn.IsNull());
+    }
+    {
+      SCOPED_TRACE("Invalid Catalog HRN (missing catalog name)");
+      HRN hrn("hrn:here:data::test:");
+      EXPECT_TRUE(hrn.IsNull());
+    }
+  }
+  {
+    SCOPED_TRACE("Schema HRN Tests");
```
Comment (andescu) [inline olp-cpp-sdk-core/tests/client/HRNTest.cpp]: Separate into new test case.
Probably you can do a separation based on ServiceType.

File: olp-cpp-sdk-core/tests/client/HRNTest.cpp [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+TEST(HRNTest, ToString) {
+  EXPECT_EQ(HRN("hrn:here:data:::hereos-internal-test-v2").ToString(),
+            "hrn:here:data:::hereos-internal-test-v2");
+  EXPECT_EQ(HRN("hrn:here:schema:::group_id:artifact_id:version").ToString(),
+            "hrn:here:schema:::group_id:artifact_id:version");
+  EXPECT_EQ(HRN("hrn:here:data:::test_pipeline").ToString(),
+            "hrn:here:data:::test_pipeline");
+}
+
+}  // namespace
```
Comment (andescu) [inline olp-cpp-sdk-core/tests/client/HRNTest.cpp]: Newline missing.

File: olp-cpp-sdk-core/tests/client/HRNTest.cpp [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+
+TEST(HRNTest, ToString) {
+  EXPECT_EQ(HRN("hrn:here:data:::hereos-internal-test-v2").ToString(),
+            "hrn:here:data:::hereos-internal-test-v2");
+  EXPECT_EQ(HRN("hrn:here:schema:::group_id:artifact_id:version").ToString(),
+            "hrn:here:schema:::group_id:artifact_id:version");
+  EXPECT_EQ(HRN("hrn:here:data:::test_pipeline").ToString(),
+            "hrn:here:data:::test_pipeline");
+}
+
```
Comment (andescu) [inline olp-cpp-sdk-core/tests/client/HRNTest.cpp]: I miss tests for parsing. Means creating a HRN and then test that each element, e.g. partition, region, account, etc., is parsed correctly.

File: olp-cpp-sdk-core/tests/client/HRNTest.cpp [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+  }
+  {
+    SCOPED_TRACE("Invalid HRN 1");
+    HRN hrn("invalid_hrn");
+    EXPECT_TRUE(hrn.IsNull());
+  }
+  {
+    SCOPED_TRACE("Invalid HRN 2");
+    HRN hrn("hrn:invalid_hrn");
+    EXPECT_TRUE(hrn.IsNull());
```
[the remark this answers is not in the document]
Comment (mykhailo-kuchma) [reply olp-cpp-sdk-core/tests/client/HRNTest.cpp]: renamed the scope, should be a separated case with partially valid hrn

File: olp-cpp-sdk-core/tests/client/HRNTest.cpp [hunk 7]
```diff
@@ -0,0 +1,192 @@
...
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+#include <olp/core/client/HRN.h>
+
+#include <gtest/gtest.h>
+#include <future>
```
Comment (borzun) [inline olp-cpp-sdk-core/tests/client/HRNTest.cpp]:
```suggestion
```
Looks like this is not needed

File: olp-cpp-sdk-core/src/client/HRN.cpp [hunk 8]
```diff
@@ -51,62 +51,73 @@ std::string HRN::ToString() const {
...
-    else if (serviceStr.compare("pipeline") == 0)
-      this->service = ServiceType::Pipeline;
-    else {
-      this->service = ServiceType::Unknown;
+  if (tokenizer.HasNext()) {
+      partition = tokenizer.Next();
+  }
+
+  if (tokenizer.HasNext()) {
+    auto serviceStr = tokenizer.Next();
```
Comment (borzun) [inline olp-cpp-sdk-core/src/client/HRN.cpp]:
```suggestion
    auto service_str = tokenizer.Next();
```
Please, change the naming of variables in this method according to Google C++ style

File: olp-cpp-sdk-core/src/client/HRN.cpp [hunk 9]
```diff
@@ -30,18 +30,18 @@ std::string HRN::ToString() const {
   std::ostringstream ret;
   std::string genericPart = ":" + region + ":" + account + ":";
```
Comment (borzun) [inline olp-cpp-sdk-core/src/client/HRN.cpp]:
```suggestion
  std::string generic_part = ":" + region + ":" + account + ":";
```

File: olp-cpp-sdk-core/src/client/HRN.cpp [hunk 10]
```diff
@@ -30,18 +30,18 @@ std::string HRN::ToString() const {
...
   ret << "hrn:" << partition << ":";
-  switch (this->service) {
+  switch (service) {
     case ServiceType::Data: {
-      ret << "data" << genericPart << this->catalogId;
-      if (!this->layerId.empty()) ret << ":" << this->layerId;
+      ret << "data" << genericPart << catalogId;
+      if (!layerId.empty()) ret << ":" << layerId;
```
Comment (borzun) [inline olp-cpp-sdk-core/src/client/HRN.cpp]:
```suggestion
      if (!layerId.empty()) { 
          ret << ":" << layerId;
      }
```

File: olp-cpp-sdk-core/src/client/HRN.cpp [hunk 11]
```diff
@@ -51,62 +51,73 @@ std::string HRN::ToString() const {
 }
 
 std::string HRN::ToCatalogHRNString() const {
-  if (this->service != ServiceType::Data) return std::string();
+  if (service != ServiceType::Data) return std::string();
```
Comment (borzun) [inline olp-cpp-sdk-core/src/client/HRN.cpp]:
```suggestion
  if (service != ServiceType::Data) {
      return std::string();
  }
```

File: olp-cpp-sdk-core/src/client/HRN.cpp (hunk 1 above)
Comment (borzun) [reply olp-cpp-sdk-core/src/client/HRN.cpp]: @mykhailo-kuchma  Why you didn't add braces here, but in other cases you did?

**Pull request merged**